pkg/client/cache/fifo.go

```diff
@@ -35,6 +35,8 @@ type FIFO struct {
 	// keyFunc is used to make the key used for queued item insertion and retrieval, and
 	// should be deterministic.
 	keyFunc KeyFunc
+	// isOlder is used to accept or reject objects which are older than what's already stored.
+	isOlder IsOlder
```

Comment:
I don't like "IsOlder", partially due to the "Is" and partially to it being a negation (if older is false continue).  "Replaces" or "CanReplace" might be more accurate, and the default is "true" because any FIFO based on an event watch is receiving events in the correct order (older -> newer)